src/config.ts

```diff
@@ -1,30 +1,17 @@
 import {
   IntegrationExecutionContext,
-  IntegrationValidationError,
-  IntegrationInstanceConfigFieldMap,
   IntegrationInstanceConfig,
+  IntegrationInstanceConfigFieldMap,
+  IntegrationValidationError,
 } from '@jupiterone/integration-sdk-core';
 import { createAPIClient } from './client';
 
-/**
- * A type describing the configuration fields required to execute the
- * integration for a specific account in the data provider.
- *
- * When executing the integration in a development environment, these values may
- * be provided in a `.env` file with environment variables. For example:
- *
- * - `CLIENT_ID=123` becomes `instance.config.clientId = '123'`
- * - `CLIENT_SECRET=abc` becomes `instance.config.clientSecret = 'abc'`
- *
- * Environment variables are NOT used when the integration is executing in a
- * managed environment. For example, in JupiterOne, users configure
- * `instance.config` in a UI.
- */
 export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
```

Comment:
Looks like the integration config map does not include the OAuth scopes. I imagine that we'll add more scopes in the future and we'll want these scopes to be labeled as "optional" for backwards compatibility. Users should be able to "opt-in" to specific integration ingestion via OAuth scopes, and we should conditionally disable steps in `getStepStartStates` using the provided OAuth scopes. 

I also noticed in the HubSpot documentation that there is a difference between required scopes and optional scopes:

> After choosing an account, you'll be presented with a list of scopes based on the &scopes= and &optional_scope= parameters you set for the authorization URL. Note: If you include an optional_scope and the selected account doesn't have access to it (such as the content scope for a CRM-only account), it will not be listed. Click the “Grant access” button to authorize the connection.

See: https://developers.hubspot.com/docs/api/creating-an-app

Wondering whether we should make everything an optional scope unless it is actually required by the integration to run. Then we should add documentation around what scopes are required vs. optional.

Our Slack integration uses the step enablement pattern based on OAuth scopes: https://github.com/JupiterOne/graph-slack/blob/master/src/getStepStartStates.ts#L48